### PR TITLE
Added possibility to add file name

### PR DESCRIPTION
**Pull request by olehmartynets**
I added possibility change file name for base64 file on android.
I do not know java and i done all what i could.
if you can , please added possibility change file name for all files and platform

### CODE REVIEW

**Comment by MateusAndrade** [comment]
Thanks for helping us with that @olehmartynets. Giving a quick look at this, i dont find how we can pass the file name from the react-native land. This still a WIP PR?

If you need any help with this, i would be glad to help you finishing this. :smile:

**Comment by olehmartynets** [comment]
@MateusAndrade 
   Share.open({
      title: _title,
      url: base64file,
      subject: title,
      filename: 'test' , // only for android
    }) 
this is example how we can pass file name from react-native

**Comment by MateusAndrade** [comment]
HI @olehmartynets  you can update your branch with the last changes on master?

**Comment by olehmartynets** [comment]
@MateusAndrade yes) i merged master in my branch

**Comment by MateusAndrade** [comment]
> @MateusAndrade yes) i merged master in my branch

Thanks! Looking at your changes and it looks fine to me. You can update our docs warning about this feature? Since this is only available on Android, it can lead to a misunderstanding for some folks.

Also you gave a look at implementing this on IOS?

**Comment by olehmartynets** [comment]
@MateusAndrade  I updated documentation.
I don't know how add this possibility for ios.
But on ios we have another way for change filename.

**Comment by MateusAndrade** [comment]
> But on ios we have another way for change filename.

Thanks for updating the docs. I will give a try at that tomorrow, and merge this ASAP. :checkered_flag:

**Comment by latusdenis** [comment]
@MateusAndrade Could you please merge it?

File: android/src/main/java/cl/json/ShareFile.java [hunk 1]
```diff
@@ -104,12 +107,13 @@ public Uri getURI() {
 
         if(this.isBase64File()) {
             String encodedImg = this.uri.getSchemeSpecificPart().substring(this.uri.getSchemeSpecificPart().indexOf(";base64,") + 8);
+            String filename = this.filename != null ? this.filename : System.nanoTime() + "";
```
Comment (jgcmarins) [inline android/src/main/java/cl/json/ShareFile.java]: what happens if user pass a fileName with special characters?

Comment (olehmartynets) [reply android/src/main/java/cl/json/ShareFile.java]: @jgcmarins you will have fileName with special characters

**Comment by latusdenis** [comment]
@jgcmarins Could this feature be merged? I've tested it in the fork and it worked perfectly, and I really need this feature in production, but without forking the library

**Comment by MateusAndrade** [comment]
@latusdenis i just merged this, please let me know if you found any problem related to this

**Comment by latusdenis** [comment]
@MateusAndrade Thank you so much, works perfectly! The only issue I see is a missing TypeScript property in Options interface. I'll with a PR on this soon.
image

**Comment by MateusAndrade** [comment]
@latusdenis i sended a PR to DefinitelyTyped that implements the new ```filename``` prop.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/38646

**Comment by darrenyaoyao** [comment]
Could you publish this version?

**Comment by VolkerSchiewe** [comment]
Can this feature be backported to version 1.x so that it can be used with react native < 0.60?

**Comment by mikehardy** [comment]
@VolkerSchiewe use jetifier in -r mode

**Pull request merged**